### PR TITLE
DEV9: Dev9 cleanup part 1.

### PR DESCRIPTION
**Pull request by lightningterror**
### Description of Changes

DEV9: Dev9 cleanup part 1.
Cleans up initializations, functions, casts and such.
### Rationale behind Changes

Cleaner code.
### Suggested Testing Steps

DEV9 works as expected, CI passes.

### CODE REVIEW

File: pcsx2/DEV9/ATA/ATA_Transfer.cpp [hunk 1]
```diff
@@ -235,8 +235,6 @@ void ATA::IO_SparseCacheLoad()
 		hddSparseBlockValid = true;
 #if defined(PCSX2_DEBUG) || defined(PCSX2_DEVBUILD)
 
-		// Store file pointer.
-		const s64 orgPos = FileSystem::FTell64(hddImage);
 		pxAssert(orgPos != -1);
```
Comment (refractionpcsx2) [inline pcsx2/DEV9/ATA/ATA_Transfer.cpp:238]: the fetching orgPos was there to fire the asserts, so we knew if the hddImage was not found. So either get rid of the assert or put it back, either or.

File: pcsx2/DEV9/ATA/ATA_Transfer.cpp [hunk 2]
```diff
@@ -282,8 +280,6 @@ void ATA::IO_SparseCacheLoad()
 			hddSparseBlockValid = true;
 #if defined(PCSX2_DEBUG) || defined(PCSX2_DEVBUILD)
 
-			// Store file pointer.
-			const s64 orgPos = FileSystem::FTell64(hddImage);
```
Comment (refractionpcsx2) [inline pcsx2/DEV9/ATA/ATA_Transfer.cpp:286]: as above

File: pcsx2/DEV9/Win32/tap-win32.cpp [hunk 3]
```diff
@@ -400,14 +400,13 @@ bool TAPGetWin32Adapter(const std::string& name, PIP_ADAPTER_ADDRESSES adapter,
 	//Note that we append to the collection during iteration
 	for (size_t vi = 0; vi < searchList.size(); vi++)
 	{
-		int targetIndex = searchList[vi];
-
 		for (ULONG i = 0; i < table->NumEntries; i++)
 		{
+			const int targetIndex = searchList[vi];
```
Comment (refractionpcsx2) [inline pcsx2/DEV9/Win32/tap-win32.cpp:405]: constant value, shouldn't really be in the loop (searchList doesn't change in this loop) else you're constantly creating and destroying it.

File: pcsx2/DEV9/ATA/ATA_Transfer.cpp (hunk 1 above)
Comment (TheLastRar) [reply pcsx2/DEV9/ATA/ATA_Transfer.cpp:238]: This line also caught my attention, and had to double check this.

Yes I added this for an assert, but this definition of orgPos ends up shadowing another definition of orgPos a few lines up (which should already have the same value)

Comment (refractionpcsx2) [reply pcsx2/DEV9/ATA/ATA_Transfer.cpp:238]: Cool then in that case the assert can be moved up (unless there's one already there, don't have the code open) and the others removed :)

Comment (TheLastRar) [reply pcsx2/DEV9/ATA/ATA_Transfer.cpp:238]: The assert should only run when the platform specific file apis (`DeviceIoControl` or `lseek`) say that part of the file is sparse, That wouldn't be the case if moved up.

That being said, I'm sure this assert could still be deduplicated

File: pcsx2/DEV9/ATA/ATA_State.cpp [hunk 4]
```diff
@@ -528,19 +528,19 @@ void ATA::HDD_SetLBA(s64 sectorNum)
 	{
 		if (!lba48)
 		{
-			regSelect = (u8)((regSelect & 0xf0) | (int)((sectorNum >> 24) & 0x0f));
-			regHcyl = (u8)(sectorNum >> 16);
-			regLcyl = (u8)(sectorNum >> 8);
-			regSector = (u8)(sectorNum);
+			regSelect = static_cast<u8>((regSelect & 0xf0) | (int)((sectorNum >> 24) & 0x0f));
```
[the remark this answers is not in the document]
Comment (lightningterror) [reply pcsx2/DEV9/ATA/ATA_State.cpp:531]: It pulled a stealth move on me.

File: pcsx2/DEV9/ATA/ATA_Transfer.cpp (hunk 1 above)
Comment (stenzek) [reply pcsx2/DEV9/ATA/ATA_Transfer.cpp:238]: Why not just `pxAssert(FileSystem::FTell64(hddImage) != -1);`?

FTell64 won't have any side-effects on the stream, so the lack of call in release should be fine.

File: pcsx2/DEV9/Win32/pcap_io_win32.cpp [hunk 5]
```diff
@@ -77,12 +77,11 @@ bool load_pcap()
 		return true;
 
 	//Store old Search Dir
-	wchar_t* oldDllDir;
 	int len = GetDllDirectory(0, nullptr);
 	if (len == 0)
 		return false;
 
-	oldDllDir = new wchar_t[len];
+	wchar_t* oldDllDir = new wchar_t[len];
```
Comment (CookiePLMonster) [inline pcsx2/DEV9/Win32/pcap_io_win32.cpp:84]: This is unrelated but now that old Windowses are unsupported, this Set/Reset dance should really be replaced with AddDllDirectory + alternate search paths + RemoveDllDirectory. Might be worth looking into for a future PR.

File: pcsx2/DEV9/ATA/ATA_Transfer.cpp (hunk 1 above)
Comment (lightningterror) [reply pcsx2/DEV9/ATA/ATA_Transfer.cpp]: All good now or you want the assert placed in a different place?

Comment (stenzek) [reply pcsx2/DEV9/ATA/ATA_Transfer.cpp]: Ugh, so you are keeping the variable around. So then you should assert on the variable, not call it again (I assumed the variable was being removed).

The whole thing's a bit iffy tbh. Later in the function, it has error handling for the tell fail case (but tell should generally never fail), but also asserts on it...

Comment (lightningterror) [reply pcsx2/DEV9/ATA/ATA_Transfer.cpp]: Yes we are keeping it, I found it odd as well that you suggested the above change, anyway should be fine now.

**Pull request merged**